cmd/api/logging/logging.go

```diff
@@ -0,0 +1,108 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logging
+
+import (
+	"bufio"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/kubearchive/kubearchive/cmd/api/abort"
+	"github.com/ohler55/ojg/jp"
+	"github.com/ohler55/ojg/oj"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+const (
+	userKey     string = "USER"
+	passwordKey string = "PASSWORD"
+	jsonPathKey string = "JSON_PATH"
+)
+
+// LogRetrieval retrieves a middleware function that checks the logging config and when set
+// expects to find a log url in the context from where retrieve and parse logs
+func LogRetrieval(loggingConfig map[string]string) gin.HandlerFunc {
+	// FIXME For now the queries to the logging backend server are done insecurely
+	client := http.Client{
+		Transport: otelhttp.NewTransport(&http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}})}
+	user := loggingConfig[userKey]
+	password := loggingConfig[passwordKey]
+	jsonPathParser, err := jp.ParseString(loggingConfig[jsonPathKey])
+	if user == "" || password == "" || err != nil {
+		return func(c *gin.Context) {
+			abort.Abort(c, fmt.Errorf("invalid jsonPath expression in logging config: %s", err.Error()),
```

Comment:
This error doesn't make sense if it's actually the `USERNAME` or `PASSWORD`  that is empty.

Reply:
fixed